Namitha: Fix calendar visibility, booked date indicators, and dark mode implementation

Namitha7070 opened the pull request:
# Description
(PRIORITY Medium) Jae/Sudarshan- Fix Calendar Not Showing on Activities Registration Page , Highlight Booked Days and Fix Dark Mode text and colours

1. Ensure the calendar component renders on the main page.
2. Fix any missing data or props that prevent the calendar from loading.
3. Verify calendar functions such as date navigation and event display work properly.
4. Check styling or layout issues that may be hiding the calendar.
5. Add a clear visual indicator (e.g., highlight, badge, or icon) on days that are booked so users can easily see which dates are already taken.
6. Fix the dark mode implementation of text fields

## Related PRS (if any):
…

## Main changes explained:

1. Calendar: Enabled react-calendar on the Register page with selection and navigation.
2. Booked Days: Highlighted with a red dot and light-red background; added a small legend.
3. Dark Mode: Used Redux [darkMode]
4. CSS Modules: Converted page styles
5. React Calendar Styles: Used :global(...) in CSS modules so calendar classes style correctly.
6. Responsiveness: Added wrap and media queries to prevent calendar cropping on small screens.
7. Lint Fixes: Removed duplicate CSS selectors to satisfy pre-commit stylelint.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ /communityportal/activities/Register/1
6. verify whether the Calendar renders properly in the page and also test the Dark Mode.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/69db6896-bb2b-4774-b89b-41f390f708d6

Namitha7070 committed:
Fix calendar visibility, booked date indicators, and dark mode implementation

Anusha-Gali reviewed (changes requested):
Hi Namitha,

I have reviewed your PR locally, the calendar does display as per requirement in both modes. But i was unable to verify the point "Verify calendar functions such as date navigation and event display work properly" and also what about if dates are available to book?
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 58 29 PM" src="https://github.com/user-attachments/assets/2e736554-5df2-471e-9b05-ad469c79708e" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 58 40 PM" src="https://github.com/user-attachments/assets/f27268b2-580b-40dc-a26a-858533967282" />

rohanrastogi311 reviewed (approved):
Hi Namitha, 

Well done with this implementation. 

<img width="1727" height="1036" alt="PR 4544 Screenshot" src="https://github.com/user-attachments/assets/6ab0f860-13eb-45f9-80f6-00a7e1dcde69" />
<img width="1728" height="1037" alt="PR 4544 Screenshot 2" src="https://github.com/user-attachments/assets/1e9fbdc6-d62c-48d7-8396-957a0bb1753a" />

HemanthNidamanuru reviewed (changes requested):
Hi Namitha,

Thanks for working on this. Here's what I found after testing locally:

Working:

1. The calendar is rendering correctly on the main page.
2. Date navigation is working as expected.

Not working / needs attention:

1. I wasn't sure how to verify the event display properly. Could you share some steps or context on how to test this?
2. Dates where tickets are available need some kind of visual indication so users can easily tell them apart from fully booked or unavailable days.

<img width="1901" height="971" alt="Image" src="https://github.com/user-attachments/assets/2015200e-1aa1-441c-8ea0-6c91aa170e77" />

Namitha7070 committed:
Fix merge conflicts and add available date indicators to calendar

Namitha7070 committed:
Fix prettier formatting errors

HemanthNidamanuru reviewed (changes requested):
Hi Namitha,

Thank you for working on the improvements. one issue found during testing:

1. Legend colors (Available, Today) are not visible in dark mode.

<img width="1888" height="1014" alt="Image" src="https://github.com/user-attachments/assets/8704822c-fae8-4175-bbd5-e4a56c5be663" />

<img width="1890" height="1028" alt="Image" src="https://github.com/user-attachments/assets/215c070e-d1de-473f-a323-2d229b86c202" />

<img width="1895" height="1039" alt="Image" src="https://github.com/user-attachments/assets/1523e038-8bd3-4516-afdb-67ca77ac0dac" />

Namitha7070 committed:
Fix calendar legend colors visibility in dark mode

Namitha7070 committed:
fix: resolve prettier formatting errors in Register.jsx

Namitha7070 committed:
fix: add PropTypes validation to DescriptionSection in EventDescription.jsx

Namitha7070 committed:
fix: replace array-index keys in EventDescription lists

Namitha7070 committed:
fix: resolve all 20 SonarQube quality gate violations

- Replace window with globalThis.window (9 occurrences)
- Replace String#replace() with String#replaceAll() for simple strings
- Update EventDescription.jsx Q: and A: prefix removal
- Update Register.jsx token parsing and localStorage access

Namitha7070 committed:
fix: correct globalThis usage - remove .window property

- Replace globalThis.window with globalThis (9 occurrences)
- globalThis.localStorage instead of globalThis.window.localStorage
- globalThis.atob instead of globalThis.window.atob
- globalThis.location instead of globalThis.window.location
- Fixes SonarQube maintainability violations

Namitha7070 committed:
fix: resolve SonarQube issues and restore yarn.lock

- Restore yarn.lock (accidentally truncated in previous commit)
- Number.parseInt over parseInt (Register.jsx L164)
- Optional chain: activity?.startTime (Register.jsx L180)
- replaceAll over replace for regex patterns
- Extract nested ternary into if/else block
- Extract tab render helpers to reduce cognitive complexity in EventDescription.jsx

Namitha7070 committed:
Merge branch 'development' into namitha/calendar-fix-dark-mode

- Resolve yarn.lock merge conflict by accepting development version
- Merge latest changes from development branch

HemanthNidamanuru reviewed (dismissed):
Approved

<img width="1890" height="953" alt="Image" src="https://github.com/user-attachments/assets/408a886b-3393-4452-8fd9-a26b82b60932" />

Namitha7070 committed:
Merge branch 'development' into namitha/calendar-fix-dark-mode

# Conflicts:
#	src/components/CommunityPortal/Activities/Register/Register.jsx
#	src/components/CommunityPortal/Activities/Register/Register.module.css

Namitha7070 committed:
Fix quality gate issues: reliability bug and stylelint errors

Namitha7070 committed:
Fix 14 accessibility contrast issues for WCAG AA compliance

Namitha7070 committed:
Fix all 12 remaining SonarQube issues: contrast, array cloning, commented code
